common.py

```diff
@@ -114,14 +148,60 @@ class TftAggregateOutput:
         plugins: a list of objects derivated from type PluginOutput for each optional plugin to append
         resulting output to."""
 
-    flow_test: IperfOutput = None
+    flow_test: Optional[IperfOutput] = None
     plugins: List[PluginOutput] = field(default_factory=list)
 
+    def __post_init__(self) -> None:
+        if isinstance(self.flow_test, dict):
+            self.flow_test = dataclass_from_dict(IperfOutput, self.flow_test)
+        elif self.flow_test is not None and not isinstance(self.flow_test, IperfOutput):
+            raise ValueError("flow_test must be an IperfOutput instance or a dict")
+
+        self.plugins = [
+            (
+                dataclass_from_dict(PluginOutput, plugin)
+                if isinstance(plugin, dict)
+                else plugin
+            )
+            for plugin in self.plugins
+        ]
+
 
-def j2_render(in_file_name, out_file_name, kwargs):
+def j2_render(in_file_name: str, out_file_name: str, kwargs: Dict[str, Any]) -> None:
     with open(in_file_name) as inFile:
         contents = inFile.read()
     template = jinja2.Template(contents)
     rendered = template.render(**kwargs)
     with open(out_file_name, "w") as outFile:
         outFile.write(rendered)
+
+
+def serialize_enum(
+    data: Union[Enum, Dict[Any, Any], List[Any], Any]
+) -> Union[str, Dict[Any, Any], List[Any], Any]:
+    if isinstance(data, Enum):
+        return data.name
+    elif isinstance(data, dict):
+        return {k: serialize_enum(v) for k, v in data.items()}
+    elif isinstance(data, list):
+        return [serialize_enum(item) for item in data]
+    else:
+        return data
+
+
+T = TypeVar("T")
+
+
+def dataclass_from_dict(cls: Type[T], data: Dict[str, Any]) -> T:
```

Comment:
Lets add a comment explaining how this function works as well

Reply:
[Sure thing](https://github.com/wizhaoredhat/ocp-traffic-flow-tests/pull/24/files#diff-306e28897a4282ae3852eb0568483edafeed04a979ec5b8afa2b2ee685f2148dR195)